### PR TITLE
Expose BigQuery Failed inserts

### PR DESCRIPTION
**Pull request by regadas**

### CODE REVIEW

**Comment by anish749** [comment]
This looks great :) Thanks for the prompt fix 👍

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/dynamic/syntax/SCollectionSyntax.scala [hunk 1]
```diff
@@ -63,7 +66,13 @@ final class DynamicBigQueryOps[T](private val self: SCollection[T]) extends AnyV
       if (writeDisposition != null) {
         transform = transform.withWriteDisposition(writeDisposition)
       }
-      self.applyInternal(transform)
+      val wr = self.applyInternal(transform)
+      if (self.internal.isBounded() == IsBounded.UNBOUNDED) {
+        insertErrorTransform(self.context.wrap(wr.getFailedInsertsWithErr()))
```
Comment (anish749) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/dynamic/syntax/SCollectionSyntax.scala]: I haven't tested this but from what I read, I think we have to call [`withExtendedErrorInfo`](https://beam.apache.org/releases/javadoc/2.11.0/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.Write.html#withExtendedErrorInfo--) first to make BQ write extended info and then we can access [`getFailedInsertsWithErr`](https://beam.apache.org/releases/javadoc/2.11.0/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.html#getFailedInsertsWithErr--)

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala [hunk 2]
```diff
@@ -171,7 +172,7 @@ final case class BigQueryTable(tableSpec: String) extends BigQueryIO[TableRow] {
     Reads.bqReadTable(sc)(beam.BigQueryIO.readTableRows(), table)
 
   override def write(data: SCollection[TableRow], params: WriteP): Tap[TableRow] = {
-    var transform = beam.BigQueryIO.writeTableRows().to(table)
+    var transform = beam.BigQueryIO.writeTableRows().to(table).withExtendedErrorInfo()
```
Comment (anish749) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala]:
```suggestion
    var transform = beam.BigQueryIO.writeTableRows().to(table)
```

I think `transform.withExtendedErrorInfo()` should be only set when `params.extendedErrorInfo` is `Enabled`?

And it is there in line 195.

Comment (regadas) [reply scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala]: Ahh good catch! that wasn't supposed to be there 😄

**Comment by jto** [comment]
@regadas can you resolve the conflicts ?

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/dynamic/syntax/SCollectionSyntax.scala [hunk 3]
```diff
@@ -57,13 +80,20 @@ final class DynamicBigQueryOps[T](private val self: SCollection[T]) extends AnyV
         .write()
         .to(destinations)
         .withFormatFunction(Functions.serializableFn(formatFn))
+        .withExtendedErrorInfo()
```
Comment (byungk) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/dynamic/syntax/SCollectionSyntax.scala]: If we set `withExtendedErrorInfo()` here, does it matter which `ExtendedErrorInfo` is actually passed in for `extendedErrorInfo` param?

Wouldn't it defeat the purpose of subsequent pattern match on `extendedErrorInfo: ExtendedErrorInfo`? _(Line 92-95)_:
```
transform = extendedErrorInfo match {
  case Disabled => transform
  case Enabled  => transform.withExtendedErrorInfo()
}
```

Was this an artifact of incorrect merge conflict resolution?

Comment (regadas) [reply scio-bigquery/src/main/scala/com/spotify/scio/bigquery/dynamic/syntax/SCollectionSyntax.scala]: Hi @byungk Nice catch!  not sure what really happen but probably was what you mentioned.

**Pull request merged**